jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -209,6 +211,39 @@ YUI.add('user-profile', function() {
     },
 
     /**
+      Fetches the model name from the modelName ref and then creates a model
+      then switches to it.
+      @method createAndSwitch
+      @param {Object} e The event handler from a form submission.
+    */
+    createAndSwitch: function(e) {
+      if (e && e.preventDefault) {
+        e.preventDefault();
+      }
+      // The supplied new model name needs to be valid.
+      var modelName = this.refs.modelName;
+      if (!modelName.validate()) {
+        // Only continue if the validation passes.
+        return;
+      }
+      // Because this will automatically connect to the model lets show
+      // the connecting mask right now.
+      this.props.showConnectingMask();
+      // XXX This is only for the JIMM flow.
+      this.props.env.createModel(
+        modelName.getValue(),
+        this.props.user.user,
+        data => {
+          if (data.err) {
+            // XXX Throw notification
+            console.log(err);
```

Comment:
done, although this is also going to have real error handling implemented to resolve the qa issue @bac found.